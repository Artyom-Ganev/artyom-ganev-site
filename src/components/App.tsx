import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.scss";
import Menu from "./Navigation/Menu";
import Main from "./Page/Main";
import About from "./Page/About";
import Blog from "./Page/Blog";
import Career from "./Page/Career";
import Contacts from "./Page/Contacts";
import NotFound from "./Page/NotFound";
import IState from "../interface/IState";
import { connect, DispatchProp } from "react-redux";
import { MAIN } from "../constants/Pages";

export interface IProperties extends DispatchProp {
  navigation: { activePage: string };
}

/**
 * Main application component
 */
const App = ({ dispatch, navigation }: IProperties) => (
  <div className="flexBox flexColumn">
    <Menu dispatch={dispatch} activePage={navigation.activePage} />
    <div className="components-app__background" />
    <div className="flexBox flexColumn components-app__content">
      <Switch>
        <Route exact={true} path="" component={Main} />
        <Route path="about" component={About} />
        <Route path="blog" component={Blog} />
        <Route path="career" component={Career} />
        <Route path="contacts" component={Contacts} />
        <Route component={NotFound} />
      </Switch>
    </div>
    <div className="components-app__rights">Â© 2019 All rights reserved</div>
  </div>
);

/**
 * Component state to props mapping function
 * @param {IState} state
 * @param {Object} ownProps
 */
const mapStateToProps = (
  { navigation }: IState,
  ownProps: { location: { pathname: string } }
) => {
  const { activePage } = navigation;
  if (activePage) {
    return { navigation: { activePage } };
  }
  // We need to set default page on first login
  const paths = ownProps.location.pathname.split("/");
  return { navigation: { activePage: paths[1] || MAIN.id } };
};

export default connect(mapStateToProps)(App);
