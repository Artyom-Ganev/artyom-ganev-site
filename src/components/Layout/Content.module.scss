.background {
  background-image: url("../../assets/images/rybinsk.jpg");
  background-size: cover;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: -1;
}

.container {
  padding-top: 4em;
}
