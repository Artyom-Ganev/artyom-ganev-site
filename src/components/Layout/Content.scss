.components-content {
  &__background {
    background-image: url("../../resources/rybinsk.jpg");
    background-size: cover;
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: -1;
  }

  &__wrapper {
    padding-top: 4em;
  }
}
