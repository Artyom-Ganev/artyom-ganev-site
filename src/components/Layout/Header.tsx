import React from 'react';
import Menu from '../Navigation/Menu';

/**
 * Header component
 */
const Header = () => (
  <header className='flexBox'>
    <Menu />
  </header>
);

export default Header;
