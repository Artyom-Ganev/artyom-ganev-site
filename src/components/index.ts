export { default as App, NarrowContext } from './App';
