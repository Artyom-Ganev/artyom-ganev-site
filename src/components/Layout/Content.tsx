import React, { FunctionComponent } from 'react';
import { Routes } from 'src/components/Navigation';
import styles from './Content.module.scss';

/**
 * App Content component
 */
const Content: FunctionComponent = () => (
  <>
    <div className={styles.background} />
    <div className={`flexBox flexColumn ${styles.container}`}>
      <Routes />
    </div>
  </>
);

export default Content;
