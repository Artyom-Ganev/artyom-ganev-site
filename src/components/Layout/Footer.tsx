import Container from '@material-ui/core/Container';
import React, { FunctionComponent } from 'react';
import { IClassName } from 'src/components/types';

export const RIGHTS_TEXT = 'Â© 2020 All rights reserved';

/**
 * Footer component
 */
const Footer: FunctionComponent<IClassName> = ({ className }: IClassName) => (
  <Container className={className} disableGutters fixed component='footer'>
    <span>{RIGHTS_TEXT}</span>
  </Container>
);

export default Footer;
