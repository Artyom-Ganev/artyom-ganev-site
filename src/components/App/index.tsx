import { createBrowserHistory } from 'history';
import React, { FunctionComponent, useEffect, useState } from 'react';
import { Router } from 'react-router-dom';
import { Content, Footer, Header } from 'src/components/Layout';

/**
 * Check if the page is narrow
 */
const isNarrow = (): boolean => window.innerWidth <= 800;

/**
 * Resize hook
 */
const useResize = (): boolean => {
  const [narrow, setNarrow] = useState<boolean>(isNarrow());
  useEffect(() => {
    const onResize = (): void => setNarrow(isNarrow());
    window.addEventListener('resize', onResize);
    return (): void => window.removeEventListener('resize', onResize);
  });
  return narrow;
};

export const NarrowContext = React.createContext(isNarrow());

export const history = createBrowserHistory();

/**
 * Main application component
 */
const App: FunctionComponent = () => (
  <Router history={history}>
    <div className='flexBox flexColumn'>
      <Header />
      <NarrowContext.Provider value={useResize()}>
        <Content />
      </NarrowContext.Provider>
      <Footer />
    </div>
  </Router>
);

export default App;
