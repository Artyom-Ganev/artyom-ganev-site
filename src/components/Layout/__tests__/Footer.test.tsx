import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Footer } from 'src/components/Layout';

describe('Footer', () => {
  const wrapper: ShallowWrapper = shallow(<Footer />);

  it('should render footer tag', () => {
    const footer = wrapper.find('footer');
    expect(footer).toHaveLength(1);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
