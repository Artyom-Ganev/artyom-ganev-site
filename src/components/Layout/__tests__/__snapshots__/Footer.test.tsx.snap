// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should match snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  className="className"
  component="footer"
  disableGutters={true}
  fixed={true}
>
  <span>
    Â© 2020 All rights reserved
  </span>
</WithStyles(ForwardRef(Container))>
`;
